python-avd/pyavd/schema/generate_docs/yamllinegen.py

```diff
@@ -0,0 +1,412 @@
+            return yaml.dump({self.schema._key: self.schema.default}, indent=2)
+
+    def get_restrictions(self) -> str | None:
+        """
+        Returns restrictions as inline semicolon separated strings.
+
+        Only covers generic restrictions. Should be overridden in type specific subclasses.
+        """
+        restrictions = []
+        valid_values = []
+        if getattr(self.schema, "dynamic_valid_values", None):
+            valid_values.append(f"<value(s) of {self.schema.dynamic_valid_values}>")
+
+        if getattr(self.schema, "valid_values", None):
+            valid_values.extend(self.schema.valid_values)
+
+        if valid_values:
+            if self.schema.type == "str":
+                # Add quotes around string values.
+                valid_values = [f'"{valid_value}"' for valid_value in valid_values]
+            else:
+                valid_values = [str(valid_value) for valid_value in valid_values]
+
+            restrictions.append(" | ".join(valid_values))
+
+        return "; ".join(restrictions) or None
+
+    def render_children(self) -> Generator[YamlLine]:
+        """Noop for classes without children. Override in subclasses for dict and list."""
+        yield from []
+
+
+class YamlLineGenBool(YamlLineGenBase):
+    pass
+
+
+class YamlLineGenInt(YamlLineGenBase):
+    def get_restrictions(self) -> str | None:
+        """
+        Returns restrictions as inline semicolon separated strings.
+
+        Leverages common restrictions from base class.
+        """
+        restrictions = []
+        if self.schema.min is not None or self.schema.max is not None:
+            if self.schema.min is not None and self.schema.max is not None:
+                restrictions.append(f"{self.schema.min}-{self.schema.max}")
+            elif self.schema.min is not None:
+                restrictions.append(f">={self.schema.min}")
+            elif self.schema.max is not None:
+                restrictions.append(f"<={self.schema.max}")
+
+        if base_restrictions := super().get_restrictions():
+            restrictions.extend(base_restrictions.split("; "))
+
+        return "; ".join(restrictions) or None
+
+
+class YamlLineGenStr(YamlLineGenBase):
+    def get_restrictions(self) -> str | None:
+        """
+        Returns restrictions as inline semicolon separated strings.
+
+        Leverages common restrictions from base class.
+        """
+        restrictions = []
+        if self.schema.min_length is not None or self.schema.max_length is not None:
+            if self.schema.min_length is not None and self.schema.max_length is not None:
+                restrictions.append(f"length {self.schema.min_length}-{self.schema.max_length}")
+            elif self.schema.min_length is not None:
+                restrictions.append(f"length >={self.schema.min_length}")
+            elif self.schema.max_length is not None:
+                restrictions.append(f"length <={self.schema.max_length}")
```

Comment:
repeating with the Int stuff - maybe some helper method?

Reply:
There are slight differences, so I don't it would be beneficial (for readability) to write a helper.